portability-types-transfer/src/main/java/org/dataportabilityproject/types/transfer/retry/RetryStrategyLibrary.java

```diff
@@ -42,13 +45,22 @@ public RetryStrategyLibrary(List<Pair<String, RetryStrategy>> strategyMapping,
    */
   public RetryStrategy checkoutRetryStrategy(Throwable throwable) {
     // TODO: determine retry strategy based on full information in Throwable
-    String exceptionMessage = throwable.getMessage();
-    for (Pair<String, RetryStrategy> entry : strategyMapping) {
-      String regex = entry.first;
-      if (exceptionMessage.matches(regex)) {
-        return entry.second;
+    // TODO: better logic (v2)
```

Comment:
what does better logic mean?

Reply:
I was looking for something cleaner.  Brian's suggestion helped.